### PR TITLE
Centreline mat invest

### PR DESCRIPTION
**Pull request by chmnata**
## What this pull request accomplishes:

- Creates centreline_latest as a mat view and a trigger function that looks at gis_core.centreline updates
- Create routing table and function for directional routing with latest centreline

## Issue(s) this solves:

- #884 

## What, in particular, needs to reviewed:

- the code! and if the comments are clear

## What needs to be done by a sysadmin after this PR is merged

- Nothing, I already created them 🙈

### CODE REVIEW

**Comment by radumas** [comment]
When #845 is merged you should add documentation for using these handy views and functions

File: gis/centreline/sql/create_view_routing_centreline_directional.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
+CREATE OR REPLACE VIEW gis_core.routing_centreline_directional AS
+
+SELECT
+    centreline.centreline_id,
+    concat(centreline.centreline_id, 0)::integer AS id,
```
Comment (radumas) [inline gis/centreline/sql/create_view_routing_centreline_directional.sql:5]: Unfortunately this isn't like the HERE `link_id`

image

If we need these to be unique then we could instead add 100,000,000 and 200,000,000 to the IDs

**Review by radumas — CHANGES_REQUESTED** [review]
Very nice. You have triggered innovation.

One comment on the directional IDs.

File: gis/centreline/sql/create_view_routing_centreline_directional.sql (hunk 1 above)
Comment (chmnata) [reply gis/centreline/sql/create_view_routing_centreline_directional.sql:5]: gahhh maybe I will just row_number it 🤔 edge id changing through out version should be ok right?

File: gis/centreline/sql/create_matview_centreline_intersection_point_latest.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
+CREATE MATERIALIZED VIEW gis_core.centreline_intersection_point_latest AS
+
+SELECT *
+FROM gis_core.centreline_intersection_point
```
Comment (radumas) [inline gis/centreline/sql/create_matview_centreline_intersection_point_latest.sql:4]: We should filter this to only include intersections that match centrelines we're pulling, to solve #819

**Review by radumas — CHANGES_REQUESTED** [review]

**Pull request merged**